triedb/pathdb/difflayer.go

```diff
@@ -26,6 +26,86 @@ import (
 	"github.com/ethereum/go-ethereum/trie/triestate"
 )
 
+type HashIndex struct {
```

Comment:
`HashIndex` is too general. I think `HashNodeCache` is more suitable.

Reply:
Updated.